lib/facter/ec2facts.rb

```diff
@@ -243,29 +249,63 @@ def query_autoscale_group(group_id, endpoint, access_key, secret_key, token)
     params['autoscaling_min_size'] = min_size_elem.text
   end
 
-  max_size_elem = doc.get_elements('//MinSize')[0]
+  max_size_elem = doc.get_elements('//MaxSize')[0]
   if max_size_elem.nil?
-    Facter.debug("No MinSize found for autoscaling group #{group_id}")
+    Facter.debug("No MaxSize found for autoscaling group #{group_id}")
     return nil
   else
     params['autoscaling_max_size'] = max_size_elem.text
   end
 
-  desired_cap_elem = doc.get_elements('//MinSize')[0]
+  desired_cap_elem = doc.get_elements('//DesiredCapacity')[0]
   if desired_cap_elem.nil?
-    Facter.debug("No MinSize found for autoscaling group #{group_id}")
+    Facter.debug("No DesiredCapacity found for autoscaling group #{group_id}")
     return nil
   else
     params['autoscaling_desired_capacity'] = desired_cap_elem.text
   end
 
+  elbs = []
+  elb_elem = doc.get_elements('//LoadBalancerNames/member').each do |item|
+    elbs << item.text
+  end
+  if elbs
+    params['autoscaling_elbs'] = elbs
+  end
+
+  # Get the ASG tags.
+  doc.get_elements('//Tags/member').each do |item|
+    key = item.get_elements('Key')[0].text
+    next if key.start_with?('aws:')
+    key.gsub!(':','_')
+    value = item.get_elements('Value')[0].text
+    params["autoscaling_tag_#{key}"] = value
+  end
+
   return params
 end
 
 # Look for our EC2 instance ID, then query the tags assigned to the instance.
 # If there is an autoscaling group tag (aws:autoscaling:groupName), also query
 # the autoscaling group min/max/desired size parameters.
 def check_facts
+
+  cache_file = '/tmp/ec2_facts.json'
+  if File.exist?(cache_file)
+    if ( ( Time.now - File.stat(cache_file).mtime ).to_i < [ 43200, 86400 ].sample )
```

Comment:
Instead of `[ 43200, 86400 ].sample`, how about `( 43200..86400 ).to_a.sample` ?
Spread the thundering herd out a lot more than just two 30 minute windows..?